articles/quickstart/spa/angular2/02-calling-an-api.md

```diff
@@ -69,203 +64,350 @@ To do this, add a `proxy.conf.json` file to the root of the project and populate
+        this.getTokenSilently$,
+        this.isAuthenticated$
+      );
+    })
+  );
+  // Subscribe to authentication completion observable
+  // Response will be an array of user, token, and login status
+  authComplete$.subscribe(([user, token, loggedIn]) => {
+    // Update subjects and loggedIn property
+    this.userProfileSubject$.next(user);
+    this.accessTokenSubject$.next(token);
+    this.loggedIn = loggedIn;
+    // Redirect to target route after callback processing
+    this.router.navigate([targetRoute]);
+  });
+}
+```
+
+In both of the above methods, we're using [`combineLatest` from RxJS](https://rxjs-dev.firebaseapp.com/api/index/function/combineLatest) to create an observable that combines `getUser$` and `getTokenSilently$` to emit results from both streams. We can then subscribe and set those values in the appropriate behavior subjects for easy access in our application.
+
+::: note
+For more details on the implementation, see the comments in the code snippets above.
+:::
+
+## Create an HTTP Interceptor
+
+In order to add the access token to the header of secured API requests, we'll create an [HTTP interceptor](https://angular.io/api/common/http/HttpInterceptor). Interceptors intercept and handle or modify HTTP requests or responses. Interceptors in Angular are services. 
 
-Next, create a new service that will be responsible for making API calls:
+Create an interceptor service with the following CLI command:
 
 ```bash
-ng generate service api
+ng generate service interceptor
 ```
 
-Then open `src/app/api.service.ts` and replace its contents with the following:
+Open the generated `src/app/interceptor.service.ts` file and add the following code:
 
 ```ts
 import { Injectable } from '@angular/core';
+import { HttpRequest, HttpHandler, HttpEvent, HttpInterceptor } from '@angular/common/http';
 import { AuthService } from './auth.service';
-import { HttpClient } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { filter, mergeMap, catchError } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
-export class ApiService {
-  constructor(
-    private authService: AuthService,
-    private httpClient: HttpClient
-  ) {}
-
-  async ping(): Promise<any> {
-    const client = await this.authService.getAuth0Client();
-    const token = await client.getTokenSilently();
-
-    return this.httpClient
-      .get('/api/external', {
-        headers: {
-          Authorization: `Bearer <%= "${token}" %>`
-        }
-      })
-      .toPromise();
+export class InterceptorService implements HttpInterceptor {
+
+  constructor(private auth: AuthService) { }
+
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return this.auth.accessToken$.pipe(
+      filter(token => typeof token === 'string'),
+      mergeMap(token => {
+        const tokenReq = req.clone({
+          setHeaders: { Authorization: `Bearer ${token}` }
```

Comment:
should be `setHeaders: { Authorization: `Bearer  <%= "${token}" %>` }`

Reply:
Fixed! Thanks!